doc/implicit_conversion.rdoc

```diff
@@ -0,0 +1,187 @@
+== Implicit Conversion
+
+Some Ruby methods accept one or more objects
+that can be either:
+* <i>Of a given class</i>, and so accepted as is.
+* <i>Implicitly convertible to that class</i>, in which case
+  the called method converts the object.
+
+For each of the relevant classes, the conversion is done by calling
+a specific conversion method:
+
+* Array: +to_ary+
+* Hash: +to_hash+
+* Integer: +to_int+
+* String: +to_str+
+
+=== Array-Convertible Objects
+
+An <i>Array-convertible object</i> is an object that:
+* Has instance method +to_ary+.
+* The method accepts no arguments.
+* The method returns an object +obj+ for which <tt>obj.kind_of?(Array)</tt> returns +true+.
+
+The examples in this section use method <tt>Array#replace</tt>,
+which accepts an Array-convertible argument.
+
+This class is Array-convertible:
+
+    class ArrayConvertible
+      def to_ary
+        [:foo, 'bar', baz = 2]
+      end
+    end
+    a = []
+    a.replace(ArrayConvertible.new) # => [:foo, "bar", 2]
+
+This class is not Array-convertible (no +to_ary+ method):
+
+    class NoToAry; end
```

Comment:
To me, because you previously used `ArrayConvertable`, `NoToAry` feels like it doesn't fit into the same taxonomy. While `ArrayConvertable` is talking about the behaviour, `NoToAry` is talking about the lack of a method.

I know your point is to try and make some name for something which essentially doesn't have any meaning (the point of the class is to show the behaviour of the method). But maybe I'd suggest if you wanted to stick with `NoToAry` then then `ArrayConvertible` should be called `HasToAry` or something like that.

Alternatively (and I think my preference), if you prefer `ArrayConvertible` then maybe `NotArrayConvertible` would be more appropriate?

The same logic applies to all subsequent class names, but again I don't want to shave the yak and would rather you took ownership and decide what makes the most sense.